aten/src/ATen/native/mps/operations/View.mm

```diff
@@ -706,15 +684,26 @@ static IntArrayRef updateTensorBaseShape(const Tensor& self)
             }
             if (needsScatter) {
               auto updatesType = getMPSScalarType(updates.scalar_type());
-              newCachedGraph->updatesTensor = mpsGraphUnrankedPlaceHolder(mpsGraph, updatesType);
+              if ((updatesType == MPSDataTypeUInt8 || updatesType == MPSDataTypeBool) && !is_macos_13_or_newer()) {
+                  updatesType = MPSDataTypeInt8;
+              }
+              newCachedGraph->updatesTensor = mpsGraphRankedPlaceHolder(mpsGraph, updatesType, getMPSShape(self.numel()));
               updatesTensor = newCachedGraph->updatesTensor;
               if (inputType != updatesType) {
                 updatesTensor = [mpsGraph castTensor:updatesTensor
                                               toType:inputType
                                                 name:@"castUpdatesTensor"];
               }
             }
-            newCachedGraph->outputTensor = chainViewOperation(newCachedGraph, size, stride, storage_offset, base_shape, needsScatter, needsBoolCast, updatesTensor);
+            newCachedGraph->outputTensor = chainViewOperation(
+              newCachedGraph, size, stride, storage_offset, base_shape, needsScatter, updatesTensor);
+
+            // Workaround for issues with bool input data type in macOS Monterey (fixed in macOS Ventura)
+            if (self.scalar_type() == kBool && !is_macos_13_or_newer()) {
+              newCachedGraph->outputTensor = [mpsGraph castTensor:newCachedGraph->outputTensor
```

Comment:
Technically, you don't need to cast the output. You could've removed the cast and passed the same inputDataType to the "output" Placeholder:
Placeholder outputPlaceholder = Placeholder(
      cachedGraph->outputTensor_, realInput, /*mpsShape*/nullptr, /*gatherTensorData=*/true, selfDataType);